packages/vx-xychart/src/types/data.ts

```diff
@@ -0,0 +1,66 @@
+import { AxisScaleOutput } from '@vx/axis';
+import {
+  ScaleTypeToD3Scale,
+  ScaleConfig,
+  ScaleConfigToD3Scale,
+  ScaleInput,
+  D3Scale,
+} from '@vx/scale';
+import DataRegistry from '../classes/DataRegistry';
+import { XYChartTheme } from './theme';
+
+export type Margin = {
+  top: number;
+  right: number;
+  bottom: number;
+  left: number;
+};
+
+export type LegendShape = 'rect' | 'line' | 'dashed-line' | 'circle';
+
+export interface DataRegistryEntry<
+  XScaleConfig extends ScaleConfig<AxisScaleOutput>,
+  YScaleConfig extends ScaleConfig<AxisScaleOutput>,
+  Datum
+> {
+  /** unique data key */
+  key: string;
+  /** array of data for the key. */
+  data: Datum[];
+  /** function that returns the x value of a datum. */
+  xAccessor: (d: Datum) => ScaleInput<ScaleConfigToD3Scale<XScaleConfig, AxisScaleOutput>>;
+  /** function that returns the y value of a datum. */
+  yAccessor: (d: Datum) => ScaleInput<ScaleConfigToD3Scale<YScaleConfig, AxisScaleOutput>>;
+  /** whether the entry supports mouse events. */
+  mouseEvents?: boolean;
+  /** Optionally update the xScale. */
+  xScale?: <XScale extends D3Scale<AxisScaleOutput>>(xScale: XScale) => XScale;
+  /** Optionally update the yScale. */
+  yScale?: <YScale extends D3Scale<AxisScaleOutput>>(yScale: YScale) => YScale;
+  /** Legend shape for the data key. */
+  legendShape?: LegendShape;
+}
+
+export interface DataContext<
```

Comment:
Same here. + Have to modify `DataRegistry` to take scales types instead of scaleconfig types as well.

```ts
export interface DataContext<
  XScale extends AxisScale,
  YScale extends AxisScale,
  Datum
> {
  xScale: XScale;
  yScale: YScale;
  colorScale: PickD3Scale<'ordinal', string, string>;
  width: number;
  height: number;
  innerWidth: number;
  innerHeight: number;
  margin: Margin;
  dataRegistry: DataRegistry<XScale, YScale, Datum>;
  registerData: (
    data:
      | DataRegistryEntry<XScale, YScale, Datum>
      | DataRegistryEntry<XScale, YScale, Datum>[],
  ) => void;
  unregisterData: (keyOrKeys: string | string[]) => void;
  setDimensions: (dims: { width: number; height: number; margin: Margin }) => void;
  theme: XYChartTheme;
}
```